### DISCUSSION TITLE
cy.exec vs cy.task performances

### DISCUSSION THREAD

**Discussion by mfrachet** [cypress-io/cypress#15240] [Questions and Help] [answered]
Let's imagine that I have that I want to run a node.js script called `compute-number.js` using Cypress on the server side.

Right now, it looks that I have two options:

- cy.exec
- cy.task

My question is: is there any performances differences (for architectural reasons) between running:

```js
cy.exec('node ./compute-numbers.js')
```

and

```js
// Note: this task only do: require('./compute-numbers');
cy.task('compute-numbers')
```

???

Thanks for your help

**Comment by bahmutov** [comment, marked as answer]
Ultimately I think it will matter very little, but I would think `cy.task` is faster to execute, since the plugins process only has to be started once, while `cy.exec` starts a child process every time it runs.

**Reply by mfrachet** [reply]
Thank you for this 😊 . To give a bit more context, I'm working on https://github.com/mfrachet/cypress-audit that aims to run lighthouse on the actual page of Cypress and I'm facing differences between using lighthouse in the cli and using lighthouse inside cypress.

I was wondering if there would be a big impact between using cy.exec for running the CLI or directly using lighthouse in node 😊 

Thanks for your help and quick answer 👍🏻
